pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -12,658 +12,253 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cassert>
 #include <set>
 #include <tuple>
 #include <vector>
 
+#include <iostream>
+
 #include "AdjointDiff.hpp"
+#include "IndicesUtil.hpp"
 #include "JacobianProd.hpp"
-#include "StateVector.hpp"
+#include "StateVectorBase.hpp"
+#include "StateVectorRaw.hpp"
```

Comment:
Good call on `StateVectorBase`! However, imho, I suggest keeping `StateVector` rather than replacing it by a new name  (`StateVectorRaw`); cause, there are the following sv classes and `StateVectorRaw` is technically that of `StateVector` has been used for bindings:
- `StateVectorBase`
- `StateVectorRaw` which extends `StateVectorBase` 
- `StateVectorManaged` which extends `StateVectorBase` (for the internal use)

Reply:
In fact, I have renamed `StateVector` to `StateVectorRaw`. I kept `StateVecBinding` in the first version of this PR, but after changing the register function to use some template metaprogramming, I found that `StateVectorRaw` and `StateVecBinding` are indistinguishable and just decided to export `StateVectorRaw`. This is the reason why we do not have `StateVector`.